daemon/indexing/listener/listener.js

```diff
@@ -182,9 +174,40 @@ async function runBatch(opts, context) {
   return lastLogBlock
 }
 
+async function withRetrys(fn) {
+  let tryCount = 0
+  while (true) {
+    try {
+      await fn() // Do our action.
+      return // it worked!
+    } catch (e) {
+      // Roughly double wait time each failure
+      let waitTime = Math.pow(100, 1 + tryCount / 6)
+      // Randomly jiggle wait time by 20% either way. No thundering herd.
+      waitTime = Math.floor(waitTime * (1.2 - Math.random() * 0.4))
+      // Max out at two minutes
+      waitTime = Math.min(waitTime, 2 * 60 * 1000)
+      console.log('ERROR', e)
+      console.log(`will retry in ${waitTime / 1000} seconds`)
+      tryCount += 1
+      await new Promise(resolve => setTimeout(resolve, waitTime))
+    }
+    if (tryCount > 10) {
+      console.log('Maximum number of retrys reached')
+      // Now it's up to our enviroment to restart us.
+      // Hopefuly with a clean start, things will work better
+      process.exit(1)
```

Comment:
Would throwing an uncaught exception be a safer way to exit ?
I was reading: https://nodejs.org/api/process.html#process_process_exit_code

Reply:
Just throwing here doesn't exit, since it's inside an async call. 

It feels like it should be okay, given that it will have been many minutes since we last wrote any data we care about.